runtime/src/bank.rs

```diff
@@ -6117,7 +6231,13 @@ impl Bank {
         let txs = vec![tx.into()];
         let batch = match self.prepare_entry_batch(txs) {
             Ok(batch) => batch,
-            Err(err) => return TransactionExecutionResult::NotExecuted(err),
+            // TODO : Not sure about application fees here
+            Err(err) => {
+                return TransactionExecutionResult::NotExecuted {
+                    error: err,
+                    application_fees: HashMap::new(),
```

Comment:
needs to include application fees

Reply:
Actually here we are preparing batch which failed, and if it fails,  we do not yet lock the accounts, so for me we should charge any application fees in this case.